app/src/main/java/com/example/home/preferences/PreferenceFragment.java

```diff
@@ -0,0 +1,127 @@
+    private Button mSaveButton;
+
+    private int mZipCode;
+    private String mZipCodeText;
+    private String mNoOfBedrooms;
+    private String mPropertyTypeText;
+
+    public PreferenceFragment(){
+    }
+
+    public PreferenceFragment(MainActivity mainActivity) {
+        // Required empty public constructor
+        this.mainActivity = mainActivity;
+    }
+
+    // TODO: Rename and change types and number of parameters
+    public static PreferenceFragment newInstance(String param1, String param2) {
+        PreferenceFragment fragment = new PreferenceFragment();
+        Bundle args = new Bundle();
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_preference, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        mBedroomTextView = view.findViewById(R.id.BedroomTextView);
+        bedroomAdapter = new ArrayAdapter<>(getContext(), R.layout.dropdown_item, getResources().getStringArray(R.array.NoOfBedrooms));
+
+        mBedroomTextView.setAdapter(bedroomAdapter);
+        mBedroomTextView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                mNoOfBedrooms = parent.getItemAtPosition(position).toString();
+                Toast.makeText(getContext(), "Item: "+ mNoOfBedrooms, Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        mProperyType = view.findViewById(R.id.PropertyTypeTextView);
+        propertyTypeAdapter = new ArrayAdapter<>(getContext(), R.layout.dropdown_item, getResources().getStringArray(R.array.PropertyType));
+
+        mProperyType.setAdapter(propertyTypeAdapter);
+        mProperyType.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                mPropertyTypeText = parent.getItemAtPosition(position).toString();
+
+                Toast.makeText(getContext(), "Item: "+ mPropertyTypeText, Toast.LENGTH_SHORT).show();
+            }
+        });
+        mZipcodeEditText = (EditText) view.findViewById(R.id.ZipcodeEditText);
+        mZipCodeText = mZipcodeEditText.getText().toString();
+        if(!mZipCodeText.equals("")){
+            mZipCode = Integer.parseInt(mZipCodeText);
+        }
+        mSaveButton = view.findViewById(R.id.SaveButton);
+        mSaveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
```

Comment:
Since this won't be the final approach, Let's remove this. This add some noise to the overall approach. What you can do, is hardcode the UserPreferences Object in the Stream Fragment while you implement this logic

You can add a todo to be more explicit about what you want to do:

TODO: Save preferences into the Database